### ISSUE TITLE
MST++的PSNR34.32是ensembel 以后的结果吗？

### ISSUE THREAD

**Issue by chenjiachengzzz** [caiyuanhao1998/MST-plus-plus#2]
作者您好，您的工作非常棒，令人激动!我有三个疑问，希望您能解答：
1）MST++的PSNR34.32是ensembel 以后的结果吗
2）train了300个epoch以后，是使用最后的pth来进行计算指标吗，还是根据测试集上的指标来选一个最好的模型
3）我按照您设置的参数进行训练，不知为何train了395个epoch...
 训练命令（一张3090显卡）：python train.py --method mst_plus_plus  --batch_size 20 --end_epoch 300 --init_lr 4e-4 --outf ./exp/mst_plus_plus/   --patch_size 128 --stride 8  --gpu_id 0 
[iter:395460/300000],lr=0.000092650,train_losses.avg=0.102350816
[iter:395480/300000],lr=0.000092685,train_losses.avg=0.102352098
[iter:395500/300000],lr=0.000092720,train_losses.avg=0.102352992
[iter:395520/300000],lr=0.000092755,train_losses.avg=0.102354914
[iter:395540/300000],lr=0.000092791,train_losses.avg=0.102356508
[iter:395560/300000],lr=0.000092826,train_losses.avg=0.102357790
然后，我用最后的模型跑出来的指标为：
load model from ../train_code/exp/mst_plus_plus/2022_04_23_14_46_46/net_395epoch.pth
method:mst_plus_plus, mrae:0.2037583440542221, rmse:0.029374651610851288, psnr:32.78853988647461
我感觉和您文中的指标还有一定差距，您觉得这正常吗，或者是有什么原因造成的（有别的epoch能跑到33.6左右的指标）
希望能得到您的解答！

**Comment by chenjiachengzzz** [comment]
非常感谢大佬的指点！

**Comment by caiyuanhao1998** [comment]
你的第130个epoch明明到了34.31 dB 了呀，貌似没啥问题？

2022-04-24 06:40:13 - Iter[130000], Epoch[000130], learning rate : 0.000241980, Train Loss: 0.165087610, Test MRAE: 0.171757042, Test RMSE: 0.025719024, Test PSNR: 34.308181763

**Comment by caiyuanhao1998** [comment]
你的第130个epoch明明到了34.31 dB 了呀，没问题的

2022-04-24 06:40:13 - Iter[130000], Epoch[000130], learning rate : 0.000241980, Train Loss: 0.165087610, Test MRAE: 0.171757042, Test RMSE: 0.025719024, Test PSNR: 34.308181763

**Comment by caiyuanhao1998** [comment]
你甚至都不愿意star，fork一遍我们的repo？
白嫖可不好。

**Comment by caiyuanhao1998** [comment]
你看我们又给你传数据，还帮你debug。
你都不愿意star，fork一遍我们的repo？

**Comment by chenjiachengzzz** [comment]
> 你看我们又给你传数据，还帮你debug。 你都不愿意star，fork一遍我们的repo？
哥，我错了

**Comment by zhangzheng2242** [comment]
请问对于您说到的有问题的数据
ARAD_1K_0314：原先的mat文件损坏，我们之前是把它去掉，不过数据集官方更换了一个mat文件。
ARAD_1K_0340：mat文件存在负数
ARAD_1K_0531：mat文件存在负数
是怎么处理的啊，直接删掉嘛

**Comment by linjing7** [comment]
你好，我们已经在MST-plus-plus/dataset/split_txt/train_list.txt中去掉这三个文件了，你不需要专门处理~

**Comment by Deep-imagelab** [comment]
> 你的第130个epoch明明到了34.31 dB 了呀，没问题的
> 
> 2022-04-24 06:40:13 - Iter[130000], Epoch[000130], learning rate : 0.000241980, Train Loss: 0.165087610, Test MRAE: 0.171757042, Test RMSE: 0.025719024, Test PSNR: 34.308181763

我这边跑了两次您的开源代码，也是无法Retrain出和您一样的结果，甚至没有一个epoch的Test MRAE达到了0.16的数量级，想问下作者您在跑模型的时候也有这么大的随机性吗

**Comment by caiyuanhao1998** [comment]
> > 你的第130个epoch明明到了34.31 dB 了呀，没问题的
> > 2022-04-24 06:40:13 - Iter[130000], Epoch[000130], learning rate : 0.000241980, Train Loss: 0.165087610, Test MRAE: 0.171757042, Test RMSE: 0.025719024, Test PSNR: 34.308181763
> 
> 我这边跑了两次您的开源代码，也是无法Retrain出和您一样的结果，甚至没有一个epoch的Test MRAE达到了0.16的数量级，想问下作者您在跑模型的时候也有这么大的随机性吗

你看我发的 log 里面的这一条

2022-04-02 19:55:01 - Iter[224000], Epoch[000224], learning rate : 0.000060919, Train Loss: 0.099466614, Test MRAE: 0.164563686, Test RMSE: 0.024763588, Test PSNR: 34.316574097

我基本上每一次都可以跑到0.16。我们是用 3090 24G 跑的。显卡不同可能跑出来结果有差别。

**Comment by Deep-imagelab** [comment]
> > > 你的第130个epoch明明到了34.31 dB 了呀，没问题的
> > > 2022-04-24 06:40:13 - Iter[130000], Epoch[000130], learning rate : 0.000241980, Train Loss: 0.165087610, Test MRAE: 0.171757042, Test RMSE: 0.025719024, Test PSNR: 34.308181763
> > 
> > 
> > 我这边跑了两次您的开源代码，也是无法Retrain出和您一样的结果，甚至没有一个epoch的Test MRAE达到了0.16的数量级，想问下作者您在跑模型的时候也有这么大的随机性吗
> 
> 你看我发的 log 里面的这一条
> 
> 2022-04-02 19:55:01 - Iter[224000], Epoch[000224], learning rate : 0.000060919, Train Loss: 0.099466614, Test MRAE: 0.164563686, Test RMSE: 0.024763588, Test PSNR: 34.316574097
> 
> 我基本上每一次都可以跑到0.16。我们是用 3090 24G 跑的。显卡不同可能跑出来结果有差别。



> > > 你的第130个epoch明明到了34.31 dB 了呀，没问题的
> > > 2022-04-24 06:40:13 - Iter[130000], Epoch[000130], learning rate : 0.000241980, Train Loss: 0.165087610, Test MRAE: 0.171757042, Test RMSE: 0.025719024, Test PSNR: 34.308181763
> > 
> > 
> > 我这边跑了两次您的开源代码，也是无法Retrain出和您一样的结果，甚至没有一个epoch的Test MRAE达到了0.16的数量级，想问下作者您在跑模型的时候也有这么大的随机性吗
> 
> 你看我发的 log 里面的这一条
> 
> 2022-04-02 19:55:01 - Iter[224000], Epoch[000224], learning rate : 0.000060919, Train Loss: 0.099466614, Test MRAE: 0.164563686, Test RMSE: 0.024763588, Test PSNR: 34.316574097
> 
> 我基本上每一次都可以跑到0.16。我们是用 3090 24G 跑的。显卡不同可能跑出来结果有差别。

我也是3090跑的，不过pytorch用的1.9.0版本
